cmds/report.c

```diff
@@ -681,8 +534,9 @@ int command_report(int argc, char *argv[], struct opts *opts)
 	if (opts->avg_total && opts->avg_self) {
 		pr_use("--avg-total and --avg-self options should not be used together.\n");
 		exit(1);
-	}
-	else if (opts->avg_total)
+	} else if (opts->fields && (opts->avg_self || opts->avg_total)) {
```

Comment:
Please follow the existing style - we put closing bracket and "else" line in separate lines.